querybook/webapp/components/DataDocScheduleList/DataDocScheduleList.tsx

```diff
@@ -102,6 +146,74 @@ const DataDocScheduleList: React.FC = () => {
     );
 
     const totalPages = Math.ceil(numberOfResults / pageSize);
+    const [showSearchFilter, setShowSearchFilter] = useState(false);
+    const filterButtonRef = useRef();
+    const dispatch = useDispatch();
+    const boards = useSelector(queryDataDocFiltersSelector);
+    useEffect(() => {
+        dispatch(fetchBoards());
+    }, []);
+
+    const handleUpdateList = React.useCallback(
+        debounce((params: OptionTypeBase[]) => {
+            updateFilters({
+                list_ids: params,
+            });
+        }, 500),
+        []
+    );
+
+    const handleUpdateStatus = React.useCallback(
+        ({ target: { value } }: { target: { value: string } }) => {
+            updateFilters({
+                status: value === '' ? null : value === 'true',
+            });
+        },
+        []
+    );
+
+    const handleUpdateScheduledOnly = React.useCallback((value) => {
+        updateFilters({
+            scheduled_only: value,
+        });
+    }, []);
+
+    const reactSelectStyle = makeReactSelectStyle(true);
+
+    const searchFiltersPickerDOM = showSearchFilter && (
+        <Popover
+            layout={['bottom', 'right']}
+            onHide={() => {
+                setShowSearchFilter(false);
+            }}
+            anchor={filterButtonRef.current}
+        >
+            <div className="DataTableNavigatorSearchFilter">
+                <div className="DataDocScheduleList_select-wrapper">
+                    <DataDocScheduleSelectionList label="Status">
+                        <SimpleSelect
+                            value={filters.status}
+                            onChange={handleUpdateStatus}
+                        >
+                            {makeSelectOptions(enabledOptions)}
+                        </SimpleSelect>
+                    </DataDocScheduleSelectionList>
+                    <DataDocScheduleSelectionList label="Lists">
+                        <Select
```

Comment:
separate this Select and the useEffect that loads boards into a BoardSelect component

Reply:
also by doing that + adding a memo, the filters can be just ids instead of options